pkg/util/metadata/metadata_test.go

```diff
@@ -148,3 +151,23 @@ func TestCheckMetaDataOpts(t *testing.T) {
 		}
 	}
 }
+
+func TestGetFromMetadataService(t *testing.T) {
+	t.Run("ignores HTTP_PROXY", func(t *testing.T) {
+		// Here I spin up an HTTP server, set it as HTTP_PROXY, and
+		// assert that the request to the Metadata server doesn't hit
+		// it.
+		fakeProxy := httptest.NewServer(http.HandlerFunc(func(_ http.ResponseWriter, _ *http.Request) {
+			t.Errorf("the call to Metadata hit the proxy server")
+		}))
+		defer fakeProxy.Close()
+
+		// defer resetting HTTP_PROXY to whatever it was before this test
+		defer func(reset string) {
+			os.Setenv("HTTP_PROXY", reset)
+		}(os.Getenv("HTTP_PROXY"))
+
+		os.Setenv("HTTP_PROXY", fakeProxy.URL)
```

Comment:
It *probably* doesn't matter, but as environment variables are global and therefore potentially 'spooky action at a distance' I think it's worth being strict here: this doesn't strictly put things back as they were. In all likelihood `HTTP_PROXY` wasn't previously set, but after execution this will leave it set to the empty string.

May I suggest a helper function something like:
```go
testSetEnv(t *testing.T, name, value string) {
  orig, ok := os.LookupEnv(name)
  t.Cleanup(func() {
    if ok {
      os.Setenv(name, orig)
    } else {
      os.Unsetenv(name)
    }
  })

  os.Setenv(name, value)
}
```

You should be able to just call this from inside the test and it will do the right thing including cleaning itself up on test exit.

Reply:
Amended the patch to unset the variable in case it was previously not set. Thank you.